helper/schema/resource_data.go

```diff
@@ -171,6 +172,14 @@ func (d *ResourceData) SetPartial(k string) {
 	}
 }
 
+func (d *ResourceData) MarkNewResource() {
+	d.isNew = true
+}
+
+func (d *ResourceData) IsNewResource() bool {
+	return d.isNew
+}
```

Comment:
In fact it doesn't seem to be equivalent, tests are failing.

It's been over a week since I had my head in this part of codebase, so I don't quite know why, but I'll find out and post an explanation here.